web/gui-v2/src/components/AddRemoveColumnDialog.jsx

```diff
@@ -0,0 +1,117 @@
+import React, { useState } from 'react';
+import { css } from '@emotion/react';
+import {
+  Checkbox,
+  Dialog,
+  DialogTitle,
+} from '@mui/material';
+
+import { ButtonStyled } from '@eto/eto-ui-components';
+
+
+const styles = {
+  columnDialog: css`
+    .MuiPaper-root {
+      background-color: var(--bright-blue-light);
+    }
+  `,
+  columnDialogContents: css`
+    font-family: GTZirkonLight;
+    padding: 0 0.5rem 0.5rem;
+    width: 320px;
+  `,
+  columnDialogTitle: css`
+    font-family: GTZirkonRegular;
+    text-align: center;
+  `,
+  columnDialogList: css`
+    background-color: var(--bright-blue-lightest);
+    display: flex;
+    flex-direction: column;
+    height: calc(100vh - 300px);
+    overflow-y: auto;
+    transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
+  `,
+  columnDialogBottom: css`
+    display: flex;
+    justify-content: center;
+    margin-top: 0.5rem;
+
+    button + button {
+      margin-left: 0.5rem;
+    }
+  `,
+  plainLabel: css`
+    color: initial;
+    font-family: GTZirkonLight;
+    font-size: initial;
+    text-transform: initial;
+  `,
+};
+
+const AddRemoveColumnDialog = ({
+  columnDefinitions,
+  isOpen,
+  selectedColumns,
+  updateIsOpen,
+  updateSelectedColumns,
+}) => {
+
+  const [columnsInternal, setColumnsInternal] = useState(() => {
```

Comment:
The internal state is so that the changes to the column selection don't happen until the user clicks the 'Apply' button.  I find it unintuitive for a modal to change the state prior to an 'Apply' button being clicked, which is what would happen if the list in the modal tied directly into the overall column selection list.

Reply:
Ah ok got it, thanks